### DISCUSSION TITLE
What does cmd = { 'TSUpdateSync', 'TSUpdate', 'TSInstall' } do?

### DISCUSSION THREAD

**Discussion by h4-m4** [nvim-treesitter/nvim-treesitter#7482] [Q&A] [answered]
I am new to neovim and have only little programming experience (I use neovim to write latex documents). For work, I use the lazyvim setup, which is a lot of fun and works well. At home, I try to start building my own config by adapting the config given by kickstart.nvim. I have encountered a small problem, which I was able to solve, but I do not understand the solution. I would like to kindly ask you to explain to me what is happening. I hope that this question is not too basic for this discussion board. 

**The problem**: I installed neovim (as described here https://github.com/neovim/neovim/blob/master/BUILD.md) and then kickstart.nvim (see https://github.com/nvim-lua/kickstart.nvim)
When opening a latex file, I get the following error message:
```
Error detected while processing BufReadPost Autocommands for "*":
Error executing lua callback: /usr/local/share/nvim/runtime/filetype.lua:35: Err
or executing lua: /usr/local/share/nvim/runtime/filetype.lua:36: BufReadPost Aut
ocommands for "*"..FileType Autocommands for "*": Vim(append):tree-sitter CLI no
t found: `tree-sitter` is not executable!
stack traceback:
        [C]: in function 'nvim_cmd'
        /usr/local/share/nvim/runtime/filetype.lua:36: in function </usr/local/s
hare/nvim/runtime/filetype.lua:35>
        [C]: in function 'nvim_buf_call'
        /usr/local/share/nvim/runtime/filetype.lua:35: in function </usr/local/s
hare/nvim/runtime/filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_buf_call'
        /usr/local/share/nvim/runtime/filetype.lua:35: in function </usr/local/s
hare/nvim/runtime/filetype.lua:10>
Press ENTER or type command to continue
```
**The fix**: When I use lazyvim instead of kickstart, opening a .tex file works without an error. In lazyvim (http://www.lazyvim.org/plugins/treesitter), the treesitter plugin is installed using the additional line 
```cmd = { 'TSUpdateSync', 'TSUpdate', 'TSInstall' },```
 which was not used in kickstart. Hence my approach to fix the error message was to copy this line to the corresponding place in my config. This fixed the error.

I have two **questions** regarding this.
1) What does "cmd" do? In :help lazy.nvim /cmd it says "Lazy-load on command", which I do not understand. 
2) How does this line fix the error message?

**Comment by clason** [comment, marked as answer]
1. You need to ask that in the LazyVim community. (And read the documentation.)

2. It means that the tree-sitter CLI is not installed; how to do that depends on your system. 

If you are starting out, I would recommend leaving nvim-treesitter uninstalled for now. (For LaTeX in particular, you are much better off installing VimTeX instead anyway.)
